### PR TITLE
added a scrollbar permanantly on the sidebar

### PR DESCRIPTION
**Pull request by fraincs**
Revamped the sidebar to make it more clear that the content is scrollable. Made the sidebar scrollable height smaller in order to have scrollbars that don't overflow the scrollable portion.

### CODE REVIEW

File: apps/docs/app/ui/layout/sidebar/sidebar.css
Comment (vicky-comeau) [inline apps/docs/app/ui/layout/sidebar/sidebar.css:1]: the scrollbar seems to be under the gradient on top? Do we want this? 
image

File: apps/docs/app/ui/layout/sidebar/sidebar.css [hunk 1]
```diff
@@ -50,17 +50,23 @@
     gap: var(--hd-space-2);
     overflow-y: scroll;
     scroll-margin-top: var(--hd-space-8);
-    overscroll-behavior: contain;
-    height: calc(100% - var(--hd-space-1));
+    scrollbar-width: thin;
+    scrollbar-color: #eaeaea transparent;
```
[the remark this answers is not in the document]
Comment (fraincs) [reply apps/docs/app/ui/layout/sidebar/sidebar.css:54]: I figured that if it was good enough for nextJS it was for us.

File: apps/docs/app/ui/layout/sidebar/sidebar.css
Comment (fraincs) [reply apps/docs/app/ui/layout/sidebar/sidebar.css:1]: Nope, I've noticed, the fix is sadly not just a z index adjustment I figured I would add this bug as a new ticket (it's already behaving like this, but its less noticeable in other situations)

File: apps/docs/app/ui/layout/sidebar/sidebar.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (fraincs) [reply apps/docs/app/ui/layout/sidebar/sidebar.css:54]: image
It's not bad either in Safari.

[the remark this answers is not in the document]
Comment (fraincs) [reply apps/docs/app/ui/layout/sidebar/sidebar.css:54]: @vicky-comeau thoughts?

**Review by vicky-comeau — APPROVED** [review]

**Pull request merged**